subsys/bluetooth/controller/ll_sw/ull_adv_sync.c

```diff
@@ -554,6 +554,151 @@ uint8_t ull_adv_sync_pdu_set_clear(struct ll_adv_set *adv,
 	return 0;
 }
 
+uint8_t ull_adv_sync_acad_enable(struct lll_adv_sync *lll_sync,
+				 uint8_t acad_len, void **acad)
+{
+	struct pdu_adv_com_ext_adv *ter_com_hdr, *ter_com_hdr_prev;
+	struct pdu_adv_ext_hdr *ter_hdr, ter_hdr_prev;
+	struct pdu_adv *ter_pdu, *ter_pdu_prev;
+	uint8_t *ter_dptr, *ter_dptr_prev;
+	uint16_t ter_len, ter_len_prev;
+	uint8_t acad_len_prev;
+	uint8_t *ad_data;
+	uint8_t ter_idx;
+	uint8_t ad_len;
+
+	/* Get reference to previous tertiary PDU data */
+	ter_pdu_prev = lll_adv_sync_data_peek(lll_sync, NULL);
+	ter_com_hdr_prev = (void *)&ter_pdu_prev->adv_ext_ind;
```

Comment:
I feel like I've asked this question before, but what was the reason to cast to a `(void *)` instead of `(struct pdu_adv_com_ext_adv *)`?

Reply:
`void *` being and is guaranteed by compilers to be assigned to pointers of other data types without explicit typecast (similar to why `malloc` returns `void *`).

Reply:
> void * being and is guaranteed by compilers to be assigned to pointers of other data types

Do you mean assignable instead of assigned? If so, that makes sense.

Reply:
yes, assignable.